9x10 Holodeck and New Holodeck Simulations

Kylerace opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Enlarges the holodeck areas on all stations to be 9x10 and adds two new simulations: space checkers and skate park. All previous holodeck simulations were changed to be 9x10 without adding anything that would break their balance. Skate park is a bunch of tables to grind on with skateboards, and space checkers is a chess board set up with white/black pawns in the checkers starting position. 

![Screenshot_137](https://user-images.githubusercontent.com/15794172/91021468-b01a4700-e5a8-11ea-83a6-b3e5e8f5a42c.png)
New Space Checkers Sim
![Screenshot_139](https://user-images.githubusercontent.com/15794172/91021487-b6102800-e5a8-11ea-8d5b-7ae56da7f464.png)
Not Shitty Space Chess
![Screenshot_128](https://user-images.githubusercontent.com/15794172/90987830-c1306d00-e542-11ea-990e-97f28115be28.png)
New Skate Park Sim 
[heres some gameplay for the skate park, which is slightly outdated](https://imgur.com/a/s1SsGES)
I'm also debating giving holodeck skateboards a smaller instability factor, so that you can do more grinding/ollies before hitting stamina crit. Currently it really doesn't take much before you're stunned
EDIT: Thanks to uomo for fixing both major bugs that made the skate park unusable 


<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Was originally started because space chess is horrible and boring, so it needed to have space for a full chess board. This PR allows all sims to have about 30 extra tiles of space, and the new sims added are very reliant on said extra space. Space checkers is just as space dependent as space chess and the skate park is still somewhat cramped even with the space. The main thing this PR does is scale everything up, which allows more people to use it at once. More people can fit in the lounge, the courtroom, both thunderdomes, etc. More of the station can rely on the holodeck for competitions: chess matches against the clown, skating matches against the captain for aa, etc. 
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
tweak: All holodeck sims now 9x10,
tweak: All stations can accommodate the enlarged holodecks
add: added skatepark sim
add: added space checkers sim
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->
All the rest of the new sims:


![Screenshot_123](https://user-images.githubusercontent.com/15794172/90580091-e0af4a80-e17c-11ea-8ad0-4fd1fc5ed48a.png)
basketball court
![Screenshot_125](https://user-images.githubusercontent.com/15794172/90698109-ccc71f80-e234-11ea-9d7c-27a8de4043c0.png)
beach
![newBEEES](https://user-images.githubusercontent.com/15794172/89702033-a5e22280-d8f1-11ea-935c-9e2a11a8ff5a.png)
anthophillia 
![newburntest](https://user-images.githubusercontent.com/15794172/89702034-a5e22280-d8f1-11ea-852b-db4f5198bf78.png)
atmospheric burn test
![Screenshot_146](https://user-images.githubusercontent.com/15794172/91110781-b0135900-e633-11ea-8e22-252caf66717e.png)
chapel courtroom
![Screenshot_142](https://user-images.githubusercontent.com/15794172/91110826-cb7e6400-e633-11ea-8feb-5c4376c58c1a.png)
dodgeball
![newemptycourt](https://user-images.githubusercontent.com/15794172/89702037-a67ab900-d8f1-11ea-91c0-c1a31e3dcaaf.png)
empty court
![newfetishschool](https://user-images.githubusercontent.com/15794172/89702038-a7134f80-d8f1-11ea-80fe-d84c154b8feb.png)
erp school sim
![newfuel](https://user-images.githubusercontent.com/15794172/89702039-a7134f80-d8f1-11ea-8cfa-78c5ac17e6d5.png)
fuel tanks
![Screenshot_97](https://user-images.githubusercontent.com/15794172/90206344-4922b400-dd98-11ea-8968-29ea6e1b285c.png)
holdout bunker
![Screenshot_116](https://user-images.githubusercontent.com/15794172/90468552-9c17a680-e0cb-11ea-9ca0-71bbf6afa969.png)
kobayashi maru
![Screenshot_144](https://user-images.githubusercontent.com/15794172/91110812-c0c3cf00-e633-11ea-8eb8-8fc286f53c92.png)
new lounge 
![Screenshot_147](https://user-images.githubusercontent.com/15794172/91110771-a7228780-e633-11ea-9922-3029d9702c90.png)
medical sim 
![newmedievaltourney](https://user-images.githubusercontent.com/15794172/89702044-a8447c80-d8f1-11ea-8489-c95e50e48b06.png)
medieval tournament
![Screenshot_145](https://user-images.githubusercontent.com/15794172/91110791-b9042a80-e633-11ea-9622-b06cbf889150.png)
picnic
![Screenshot_124](https://user-images.githubusercontent.com/15794172/90677389-2ff38a80-e212-11ea-9fc8-de484ac28c58.png)
winterwonderland (made by uomo)
![Screenshot_107](https://user-images.githubusercontent.com/15794172/90304180-d08c2800-de69-11ea-92cf-b823ead1e485.png)
target practice
![newthunderdomesim](https://user-images.githubusercontent.com/15794172/89702048-a975a980-d8f1-11ea-9456-f4434853e356.png)
thunder dome
![newwildlifesim](https://user-images.githubusercontent.com/15794172/89702049-a975a980-d8f1-11ea-9e30-59d6882df52c.png)
wildlife sim
![winterv3](https://user-images.githubusercontent.com/15794172/89721216-dbd5e400-d98f-11ea-9339-53651a54005b.png)
photobooth

New Station Holodecks:
![meta9000](https://user-images.githubusercontent.com/15794172/90468618-c2d5dd00-e0cb-11ea-8f93-aca7cddf332b.png)
Meta (moved maint to the right, and changed the shape of the airlock, the room below the holodeck is now bigger since it was merged with a hallway)
![delta9000](https://user-images.githubusercontent.com/15794172/90468643-ce290880-e0cb-11ea-823c-d8a4495d61f5.png)
Delta (not many changes, just bigger, and no adjacent carp)
![icebox9000](https://user-images.githubusercontent.com/15794172/90468660-d7b27080-e0cb-11ea-8b2e-a68f85a69a92.png)
Icebox (moved maint around a bit)
![pubby9000](https://user-images.githubusercontent.com/15794172/90468672-dd0fbb00-e0cb-11ea-9bc7-77ad93a3761d.png)
Pubby (lengthened top maint hallway to meet it, and moved bottom maint to the right)

Kylerace committed:
Merge remote-tracking branch 'upstream/master'

Kylerace committed:
Bigger Holodeck

Changes the holodeck in all stations to be 9x10 and adds the space checkers and skate park simulations.

Tlaltecuhtli commented:
idk man, making every other holodeck sim feel like an almost empty room in exchange of the chess sim no one uses to feel realistic?
20 holoeswords and 4 claymores might be a slipshod inkling

Kylerace commented:
> idk man, making every other holodeck sim feel like an almost empty room in exchange of the chess sim no one uses to feel realistic?
> 20 holoeswords and 4 claymores might be a slipshod inkling
its important to note that ALL of those disappear if anyone changes the holodeck sim, and thats assuming the safety's are off. It takes four hits with a holo esword to down someone with stamina damage, and each hit doesnt slow people down that much. holo eswords in my experience are pretty hard to use for stunning. I dont see a problem with having more holoeswords, since only one round type even has a potential use for the extra swords (assuming all antags used a holosword together, and that no one changes the sim).
However the extra claymores are pretty strong, and i should get rid of two. I assumed that all holo melee weapons only dealt stamina damage.

Kylerace committed:
Small tweaks v1

removed two claymores from 1218, added more paints to the snow one, made lounge a bit comfier, added soap to bunker

Fikou commented:
they all look more ugly empty and dumb

Tharcoonvagh commented:
Swap the dark clown and mime on the chess sim
Isnt the bee sim supposed to have a beekeper suit and hat?

Tharcoonvagh commented:
Also yeah they do look kinda empty

Kylerace commented:
The clown is the king and the mime is the queen, so i cant swap their positions without violating the chess starting position, also the bee sim does have a beekeeping suit, but for some reason it doesn't look like one
![beeeeee](https://user-images.githubusercontent.com/15794172/89706647-89a7ab00-d91c-11ea-9343-b0623f796cf1.png)

Kylerace commented:
![newerlounge](https://user-images.githubusercontent.com/15794172/89707040-b4473300-d91f-11ea-9be6-0c7a5166a768.png)
![newfiringrange](https://user-images.githubusercontent.com/15794172/89707041-b5786000-d91f-11ea-9e3f-1aedc86c3a8f.png)
![newwinterthing](https://user-images.githubusercontent.com/15794172/89707043-b7422380-d91f-11ea-8e08-c49a25849d09.png)
Here are some of the improvements im working on to make things less empty, dont really know what to do with the winter one, since its original doesnt have a whole lot of stuff to expand on, also not quite sure what to do with the chapel courtroom

ATH1909 commented:
Nice.

Could you add a second hoop to the bottom of the basketball court simulation, or can hoops only ever face south?

Kylerace commented:
Unfortunately no, there is no northwards facing hoop that I can find, there are east/west ones but since the sims are 10 tiles tall that would make the hoops off center

vkalls commented:
Someone really wanted to play real chess huh? 👀

ArcaneMusic commented:
Did you remove the maint door off of icebox holodeck?
I also can't tell, but are the rechargers still there on the kobiyashi maru? That's one of the few reasons people use that holodeck loadout.
Also, blocking a maint door by default with movables is kinda cringe tbh

Kylerace committed:
Icebox maint door, kobayashi maru recharger

Kylerace committed:
Icebox maint door, kobayashi maru recharger

Also removes canvasses blocking meta maint door

Kylerace commented:
Whoops didnt see those canvasses, thank you. Also a recharger has been re added to kobayashi maru and icebox has a maint door again.
![newermetamaint](https://user-images.githubusercontent.com/15794172/89717027-43c40480-d967-11ea-807c-02faa1db02f9.png)
![newerkobayashi](https://user-images.githubusercontent.com/15794172/89717031-4e7e9980-d967-11ea-9040-acd747ebb6ea.png)
![newericebox](https://user-images.githubusercontent.com/15794172/89717038-67874a80-d967-11ea-9c96-cfcc942f613a.png)

Kylerace committed:
Sims update, station fixes

added more to the lounge (dnd friendly?), chapel courtroom has more room filled up, winter wonderland now has pipe cleaners,paper, and a canvas, forts in snowball fight or whatever its called, captains chair in kobayashi maru, two targets in target range, and rearranged medical sim. Icebox and pubby have minor fixes to holodecks

Kylerace committed:
Merge remote-tracking branch 'origin/biggerholodeck' into biggerholodeck

Git scares me, hopefully i fixed conflicts right

Tharcoonvagh commented:
Wait im dumb aaa

trollbreeder commented:
if you make it 10x10, you can make space chess and space checkers have a whole ring around, it looks nicer and allows space for 8 more captured pieces

Kylerace commented:
Originally I did that, and it does look nicer when it comes to the checkers and chess sims, but many of the other sims were designed for there to be an odd number of horizontal tiles. For example, the chapel courtroom has the judge sit in the exact middle, the thunderdome 1218 sim has the crown in the exact middle, kobayashi maru has the front chair in the center, etc. If it was 10 tiles horizontally everything would have to be either to the left or right of the center, so I decided to make it 9x10.

GravitronBlue commented:
Mind verifying that the skateboards don't become permanant once they're 'deployed'?
Pretty concerned that their holo-quality doesn't carry over when placed.

Kylerace committed:
Removed skate park sim until bugs are fixed

Currently there are two bugs that make skate park unusable, the first is that skateboards lose hologram status if you deploy them, second is that tables created from the holodeck are invisible for some reason.

Kylerace committed:
Removed skate park sim until bugs are fixed

Currently there are two bugs that make skate park unusable, the first is that skateboards lose hologram status if you deploy them, second is that tables created from the holodeck are invisible for some reason.

Kylerace committed:
Merge remote-tracking branch 'origin/biggerholodeck' into biggerholodeck

Kylerace commented:
Unfortunately I think they do disappear, with both this and the invisible table bug the skate park is unusable. They'll have to be fixed before I can re add it

Kylerace committed:
Re-adds skatepark sim due to bugfix pr by uomo

puts the skate park back in, adds new type of holographic skateboard which cant be picked up. Also removes the dispensers from lounge because duh infinite reagents.

Kylerace commented:
Update: Massive thanks to uomu for https://github.com/tgstation/tgstation/pull/52924#issue-467665975 and for fixing the skateboard problem, so now the skating park sim can be re added.

Kylerace committed:
Re-adds cameras, small changes to simulations.

Put back in cameras to the on station holodecks, and moved around some sims because they were ugly or blocked doors. firing range is now just one target, lounge no longer has a matchbox, took the intercom out of kobayashi maru, and rearranged the medical sim.

Kylerace committed:
Re-adds cameras, small changes to simulations.

Put back in cameras to the on station holodecks, and moved around some sims because they were ugly or blocked doors. firing range is now just one target, lounge no longer has a matchbox, took the intercom out of kobayashi maru, and rearranged the medical sim.

Kylerace committed:
Merge remote-tracking branch 'origin/biggerholodeck' into biggerholodeck

Kylerace committed:
Stair fix

fixes lounge stairs

Kylerace committed:
Merge branch 'master' into biggerholodeck

Kylerace committed:
im so desperate to fucking fix this conflict

Kylerace committed:
from hell’s heart I stab at thee; for hate’s sake I spit my last breath at thee.

re adds every map + the holo parts

Kylerace committed:
More map fixes

Kylerace committed:
Destroys the centcom map conflict

Kylerace committed:
Merge branch 'master' into biggerholodeck

Kylerace committed:
Makes skate park sim more radical, changes chapel courtroom

(probably) improves the skatepark layout to get rid of unnecessary  parts and replaces chapel courtroom lockers with tables due to items within not copying over

Kylerace committed:
re-adds north facing hoop to basketball sim, other small changes

put the space checkers sim right next to space chess in the holodeck list, and removes walls from winterwonderland

Kylerace committed:
re-adds north facing hoop to basketball sim, other small changes

put the space checkers sim right next to space chess in the holodeck list, and removes walls from winterwonderland, also changes skate park again

uomo91 committed:
New winter wonderland.

Kylerace committed:
Merge pull request #1 from uomo91/biggerholodeck

New winter wonderland.

Kylerace committed:
Merge remote-tracking branch 'origin/biggerholodeck' into biggerholodeck

Kylerace committed:
Adds more cameras, changes the beach sim

All on station holodecks have a holodeck-fore camera and a holodeck-aft camera. beach sim is more filled in

Kylerace commented:
gameplay of the skate park sim

Kylerace committed:
Changes how skateboards get picked up

Suggested change by AnturK, creates a pick_up_board proc that MouseDrop calls so that the new holographic skateboard can override that instead of MouseDrop

Kylerace commented on code/modules/vehicles/scooter.dm:
```diff
@@ -149,16 +149,19 @@
-	var/mob/living/carbon/M = usr
-	if(!istype(M) || M.incapacitated() || !Adjacent(M))
+	var/mob/living/carbon/Skater = usr
+	if(!istype(Skater) || Skater.incapacitated() || !Adjacent(Skater))
```
```
/obj/vehicle/ridden/scooter/skateboard/MouseDrop(atom/over_object)
	. = ..()
	var/mob/living/carbon/Skater = usr
	if(!istype(Skater))
		return
	if (over_object == Skater)
		pick_up_board(Skater)

/obj/vehicle/ridden/scooter/skateboard/proc/pick_up_board(mob/living/carbon/Skater)
	if (Skater.incapacitated() || !Adjacent(Skater))
		return
	if(has_buckled_mobs())
		to_chat(Skater, "<span class='warning'>You can't lift this up when somebody's on it.</span>")
		return
	var/board = new board_item_type(get_turf(Skater))
	Skater.put_in_hands(board)
	qdel(src)
```

Kylerace committed:
Adds dark holofloor, makes med sim non shitty

replaces the dark squares in the chess/checkerboard sims with actual dark squares too

Kylerace committed:
redoes chapel courtroom and picnic, other fixes

fixes decals in dodgeball, changes lower left room in med sim to be green.

Kylerace committed:
Merge remote-tracking branch 'upstream/master' into biggerholodeck

Kylerace committed:
fixing map conflicts episode 2 electric boogaloo

removes some unnecessary stations from the pool

Kylerace committed:
map conflicts again

re adds every map with a copy from tgstation master

Kylerace committed:
Re adds all holodeck changes

hopefully this works

Kylerace committed:
Merge remote-tracking branch 'upstream/master' into biggerholodeck

Kylerace committed:
Changes chapel courtroom to have more fancy tables, reduces pet spawners in pet park

uomo91 committed:
Makes the pet spawner use more variety.

Kylerace committed:
Merge pull request #2 from uomo91/biggerholodeck

Makes the pet spawner use more variety.
